boards/arm/ev11l78a/ev11l78a.dts

```diff
@@ -41,6 +54,11 @@
 		sense-gain-mult = <100>;
 	};
 
+	enable_sink: enable_sink {
+		compatible = "microchip,ev11l78a-ensink";
+		gpios = <&porta 22 GPIO_ACTIVE_HIGH>;
+	};
```

Comment:
looks like a `regulator-fixed`?

Reply:
so if controlled by TCPC it should be a property there, something like:

```
tcpc {
    enable-sink-gpios =  <>;
};
```

then TCPC code would use the GPIO API. Or, if using regulators:

```
enable_sink: enable-sink {
    compatible = "regulator-fixed";
    enable-gpios = <...>;
};

...

tcpc {
    ...
    sink-supply = <&enable_sink>;
};
```

Then TCPC would use the regulators API (main advantage is that is reference counted, useful if supply is shared).

Reply:
It depends on how it is used, if by multiple clients then regulator is a good fit, is by a single client, a GPIO does the job. A typical case for GPIO usage is e.g. on sensors that have a shutdown pin. I don't know this case in detail to suggest one way or the other.

Reply:
To be honest, this is almost the literal case from [the documentation you linked earlier](https://docs.zephyrproject.org/latest/samples/basic/custom_dts_binding/README.html) as it's a GPIO connected to a specific MOSFET…

Leaving it as a generic GPIO doesn't seem right though — it's most certainly no longer "General-purpose" once it's wired to the MOSFET on the board, and it has a very specific function.